### ISSUE TITLE
scSSL-Bench：单细胞自监督学习方法比较

### ISSUE THREAD

**Issue by ixxmu** [ixxmu/mp_duty#7455]
https://mp.weixin.qq.com/s/SrjK3G-m1tmzhjPvNDpxvA

**Comment by ixxmu** [comment]
scSSL-Bench：单细胞自监督学习方法比较 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com" data-pm-slice="0 0 []"><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8XsLZa3nXe6LOVXc6r8AbcuLEnzhhmS2DwRNhKkyC7X80zFGQDmeYRCXaAde1IoqavhicnD5TAFic7aUKuIlMVrfjAdkRPK9uUuU/640?wx_fmt=png&amp;from=appmsg" data-ratio="0.29074074074074074" data-type="png" data-w="1080" data-imgfileid="100057567" data-aistatus="1" src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8XsLZa3nXe6LOVXc6r8AbcuLEnzhhmS2DwRNhKkyC7X80zFGQDmeYRCXaAde1IoqavhicnD5TAFic7aUKuIlMVrfjAdkRPK9uUuU/640?wx_fmt=png&amp;from=appmsg"></span></figure><p data-tool="mdnice编辑器"><span leaf="">单细胞数据规模越来越大，但高质量标签并不总是容易获得。很多时候，我们有大量 scRNA-seq 或 multi-omics 数据，却没有足够可靠的细胞类型标签、疾病状态标签，或者跨数据集统一标注。</span><strong><span leaf="">自监督学习</span></strong><span leaf="">的吸引力就在于，它不需要直接依赖人工标签，而是通过对同一个细胞生成不同形式的扰动数据、预测缺失信息或学习数据内部结构，得到一个可用于下游任务的细胞表示。</span></p><p data-tool="mdnice编辑器"><span leaf="">很多自监督学习方法最早来自计算机视觉，它们在图像任务中非常成功，但</span><strong><span leaf="">单细胞数据不是图像</span></strong><span leaf="">，图像增强可以裁剪、翻转、颜色扰动，而单细胞表达矩阵中的每个基因都有生物学含义，随意扰动可能会破坏真实信号。与此同时，单细胞领域也出现了很多专门方法，比如 CLEAR、CLAIRE、Concerto、scCLIP，以及 scVI、totalVI、scGPT、Geneformer、scBERT 等。</span></p><p data-tool="mdnice编辑器"><span leaf="">所以这篇文章构建了一个名为 </span><strong><span leaf="">scSSL-Bench</span></strong><span leaf=""> 的系统 benchmark，比较了 </span><strong><span leaf="">19 种方法、9 个单细胞数据集</span></strong><span leaf="">，并围绕三个常见下游任务进行评估：</span><strong><span leaf="">batch correction、cell type annotation 和 missing modality prediction</span></strong><span leaf="">。</span></p><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8XUN12dhJgHIeUZniaJIjTJ02BNzibYlb9c7nSHN91YTibJbvdCicqxDoGT64iacRe2eGicicARMhOZxE2G4xFEaqvqUNVKPXocJMKyw4/640?wx_fmt=png&amp;from=appmsg" data-ratio="0.375" data-type="png" data-w="1080" data-imgfileid="100057564" data-aistatus="1" src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8XUN12dhJgHIeUZniaJIjTJ02BNzibYlb9c7nSHN91YTibJbvdCicqxDoGT64iacRe2eGicicARMhOZxE2G4xFEaqvqUNVKPXocJMKyw4/640?wx_fmt=png&amp;from=appmsg"></span></figure><h2 data-tool="mdnice编辑器"><span><span leaf="">研究设计</span></span></h2><p data-tool="mdnice编辑器"><span leaf="">文章把方法分成了几类。</span></p><p data-tool="mdnice编辑器"><span leaf="">第一类是</span><strong><span leaf="">通用自监督学习方法</span></strong><span leaf="">，包括 SimCLR、MoCo、SimSiam、NNCLR、BYOL、VICReg 和 Barlow Twins。这些方法原本主要用于图像表示学习，文章把它们迁移到单细胞数据上。</span></p><p data-tool="mdnice编辑器"><span leaf="">第二类是</span><strong><span leaf="">单细胞专用的 contrastive learning 方法</span></strong><span leaf="">，比如 CLEAR、CLAIRE、Concerto 和 scCLIP。它们会利用单细胞数据特点设计正负样本、batch 邻居或者多模态配对。</span></p><p data-tool="mdnice编辑器"><span leaf="">第三类是</span><strong><span leaf="">生成式或 foundation model 方法</span></strong><span leaf="">，包括 scVI、totalVI、scGPT、Geneformer、scBERT、scButterfly 和 scTEL。其中 scVI 和 totalVI 是单细胞领域非常常用的 VAE 类模型，scGPT、Geneformer 和 scBERT 则属于近几年很热门的 single-cell foundation model。</span></p><p data-tool="mdnice编辑器"><span leaf="">最后还有 </span><strong><span leaf="">baseline</span></strong><span leaf="">，比如 PCA 和 SCDC。</span></p><p data-tool="mdnice编辑器"><span leaf="">数据方面，文章使用了 </span><strong><span leaf="">7 个单模态 scRNA-seq 数据集</span></strong><span leaf="">，包括 PBMC、Pancreas、Immune Cell Atlas、MCA、HIC、Lung 和 Tabula Sapiens；另外还有 </span><strong><span leaf="">2 个 CITE-seq 多模态数据集</span></strong><span leaf=""> PBMC-M 和 BMMC。这两个多模态数据同时包含 RNA 和 protein / ADT 信息，因此可以评估模型是否真的学到了跨模态关系。</span></p><h2 data-tool="mdnice编辑器"><span><span leaf="">任务一：batch correction</span></span></h2><p data-tool="mdnice编辑器"><span leaf="">文章用 </span><strong><span leaf="">Bio、Batch 和 Total</span></strong><span leaf=""> 三类指标评估结果。Bio 表示生物学结构是否保留，Batch 表示批次效应是否被去除，Total 是两者加权后的综合分数。</span></p><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8VkuSgibbmIoz5ocSiboSRGiby3XlJNXVys4fHguqIWIfLJT8Kg8rBmYNpDU4fe3ibmrNXCRKMFV7fficMN4Ed9oY9xk32E7OKibic4gk/640?wx_fmt=png&amp;from=appmsg" data-ratio="1.0240740740740741" data-type="png" data-w="1080" data-imgfileid="100057566" data-aistatus="1" src="https://mmbiz.qpic.cn/mmbiz_png/Go1sLc9TF8VkuSgibbmIoz5ocSiboSRGiby3XlJNXVys4fHguqIWIfLJT8Kg8rBmYNpDU4fe3ibmrNXCRKMFV7fficMN4Ed9oY9xk32E7OKibic4gk/640?wx_fmt=png&amp;from=appmsg"></span></figure><p data-tool="mdnice编辑器"><span leaf="">在单模态 scRNA-seq 数据上，表现最好的主要是 </span><strong><span leaf="">scVI、CLAIRE 和 finetuned scGPT</span></strong><span leaf="">。scVI 在 PBMC、Pancreas、Immune Cell Atlas 等数据上整体表现稳定，说明传统单细胞生成式模型在 batch correction 上仍然很强。CLAIRE 的 batch correction 分数很高，但有时会牺牲一部分 biological conservation，可能存在过度校正。finetuned scGPT 在部分大数据集上表现很好，但 zero-shot scGPT 和 finetuned Geneformer 在 batch correction 上并不理想。</span></p><p data-tool="mdnice编辑器"><span leaf="">而在多模态数据上，PBMC-M 和 BMMC 上表现更好的不是那些专门为单细胞多模态设计的方法，而是</span><strong><span leaf="">通用 SSL 方法</span></strong><span leaf="">，比如 SimCLR、BYOL、MoCo 和 VICReg。尤其是 </span><strong><span leaf="">VICReg 和 SimCLR</span></strong><span leaf="">，在多模态 batch correction 中表现很有竞争力。相比之下，Concerto、scCLIP、scButterfly 和 totalVI 的结果并没有稳定超过这些通用方法。这说明当前单细胞多模态自监督方法还有</span><strong><span leaf="">明显改进空间</span></strong><span leaf="">。多模态整合不是简单地把 RNA 和 protein 放进一个模型里就可以解决，模型还需要真正学到不同模态之间的对应关系，同时又不能破坏细胞类型结构。</span></p><h2 data-tool="mdnice编辑器"><span><span leaf="">任务二：cell type annotation</span></span></h2><p data-tool="mdnice编辑器"><strong><span leaf="">细胞类型注释</span></strong><span leaf="">也可以理解为 query-to-reference mapping。训练集是带标签的 reference，测试集是 hold-out batch，模型需要把 query 映射到同一个 latent space，然后用 kNN classifier 预测细胞类型。</span></p><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8W1YVwOe0x3Dy8pnPsdunmpdfWSYVFJlo2xQ1PlibKwrveapReZP2pSiaYicETrYdCcKK45M4DSDcUicns42f0dTCvqRic8ooFZDlKc/640?wx_fmt=png&amp;from=appmsg" data-ratio="0.6058631921824105" data-type="png" data-w="614" data-imgfileid="100057563" data-aistatus="1" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8W1YVwOe0x3Dy8pnPsdunmpdfWSYVFJlo2xQ1PlibKwrveapReZP2pSiaYicETrYdCcKK45M4DSDcUicns42f0dTCvqRic8ooFZDlKc/640?wx_fmt=png&amp;from=appmsg"></span></figure><p data-tool="mdnice编辑器"><span leaf="">结果显示，Immune Cell Atlas 上 </span><strong><span leaf="">finetuned scGPT、Geneformer、scBERT 和 VICReg</span></strong><span leaf=""> 都有不错表现。其中，finetuned scGPT 和 Geneformer 在这个较大的 Immune Cell Atlas 数据集上表现突出，VICReg 也非常接近。相比之下，PCA 和 zero-shot scGPT 表现较差。</span></p><p data-tool="mdnice编辑器"><span leaf="">在 Pancreas 数据集上，CLEAR 和 CLAIRE 在多个 hold-out study 中表现很好，VICReg 也很稳定。这说明在细胞类型注释任务中，</span><strong><span leaf="">一些通用 SSL 方法并不弱</span></strong><span leaf="">，甚至可以超过不少专门方法。对于多模态细胞类型注释，</span><strong><span leaf="">scButterfly 表现最好</span></strong><span leaf="">，VICReg 和 SimCLR 也非常接近。</span></p><h2 data-tool="mdnice编辑器"><span><span leaf="">任务三：missing modality prediction</span></span></h2><p data-tool="mdnice编辑器"><span leaf="">训练时模型可以看到 RNA 和 protein / ADT，但测试时只给 RNA 或 GEX，然后要求模型预测缺失的 protein / ADT。这个任务比单纯 batch correction 更能反映模型是否学到了模态之间的关系。</span></p><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8UYNe2ianibv8MKQp0nStTzfFUyL0JuCwKiaHENO2GlA0FeOVaMqC5masGLNBrTz5BianzSvNy3zEoAGoWFnPUMKs8ljXYh0hDVtbw/640?wx_fmt=png&amp;from=appmsg" data-ratio="0.5276872964169381" data-type="png" data-w="614" data-imgfileid="100057565" data-aistatus="1" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8UYNe2ianibv8MKQp0nStTzfFUyL0JuCwKiaHENO2GlA0FeOVaMqC5masGLNBrTz5BianzSvNy3zEoAGoWFnPUMKs8ljXYh0hDVtbw/640?wx_fmt=png&amp;from=appmsg"></span></figure><p data-tool="mdnice编辑器"><span leaf="">图中横轴是不同方法，纵轴是预测 protein / ADT 与真实值之间的 Pearson correlation。结果显示 </span><strong><span leaf="">VICReg 和 SimCLR 表现最好</span></strong><span leaf="">，明显超过 Concerto、scCLIP、scTEL 等单细胞专用多模态方法。scButterfly 也被评估了两种方式，一种是通过 kNN 推断，一种是直接生成 protein，但整体 Pearson correlation 仍然不如最好的通用方法。</span></p><h2 data-tool="mdnice编辑器"><span><span leaf="">数据增强：masking 反而最有用</span></span></h2><p data-tool="mdnice编辑器"><span leaf="">自监督学习离不开 </span><strong><span leaf="">data augmentation</span></strong><span leaf="">。图像里常用 crop、blur、color jitter，但单细胞数据应该怎么增强，一直没有很统一的答案。这篇文章比较了多种增强方式，包括 Gaussian noise、InnerSwap、Masking、CrossOver、MNN 和 BBKNN。其中 Masking 是随机把一部分基因表达设为 0，Gaussian noise 是加噪声，InnerSwap 是在同一个细胞内部交换部分基因表达，CrossOver 是和另一个细胞交换部分表达值。MNN 和 BBKNN 则会利用细胞邻居或 batch-balanced graph 生成增强视图。</span></p><figure data-tool="mdnice编辑器"><span leaf=""><img data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8UJISmomN1iapBicic9GbnnD79cpbDomQ7lAuN0ViaEZWlREbxPnibnt9XvnZk2frEQ9acQwBHrqUPluUzjBmcRRzNsKJ6EibNoetdm4/640?wx_fmt=png&amp;from=appmsg" data-ratio="0.7246835443037974" data-type="png" data-w="632" data-imgfileid="100057568" data-aistatus="1" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Go1sLc9TF8UJISmomN1iapBicic9GbnnD79cpbDomQ7lAuN0ViaEZWlREbxPnibnt9XvnZk2frEQ9acQwBHrqUPluUzjBmcRRzNsKJ6EibNoetdm4/640?wx_fmt=png&amp;from=appmsg"></span></figure><p data-tool="mdnice编辑器"><span leaf="">结果显示，</span><strong><span leaf="">Masking 是最稳定、最有效的增强方式</span></strong><span leaf="">。无论单独使用，还是和其他增强方式组合，masking 往往都能带来不错的效果。相比之下，MNN 和 BBKNN 这类更复杂、显式利用 batch 邻居的增强，并没有稳定超过 masking。可能的原因是，单细胞表达矩阵本身具有稀疏性和 dropout 特征，masking 更接近数据本身的不确定性，也能迫使模型学习更稳健的细胞状态表示。</span></p><h2 data-tool="mdnice编辑器"><span><span leaf="">结论</span></span></h2><p data-tool="mdnice编辑器"><span leaf="">整体看下来，在单模态 scRNA-seq 的 batch correction 里，</span><strong><span leaf="">scVI</span></strong><span leaf=""> 依然是很难绕开的强 baseline，</span><strong><span leaf="">CLAIRE</span></strong><span leaf=""> 和 </span><strong><span leaf="">finetuned scGPT</span></strong><span leaf=""> 也值得重点关注，但到了多模态整合和缺失模态预测，</span><strong><span leaf="">SimCLR</span></strong><span leaf=""> 和 </span><strong><span leaf="">VICReg</span></strong><span leaf=""> 这类通用 SSL 方法在多个任务上都很不错。</span></p><p data-tool="mdnice编辑器"><span leaf="">single-cell foundation model 的 </span><strong><span leaf="">fine-tuning 很关键</span></strong><span leaf="">，zero-shot 表现并不稳定，finetuned scGPT 在部分任务上确实表现不错，Geneformer 在某些注释任务中也有优势，但它们并不是所有场景下的默认最优解。</span></p><p data-tool="mdnice编辑器"><span leaf="">增强策略方面，</span><strong><span leaf="">masking</span></strong><span leaf="">最稳定；相比之下，一些更复杂的 batch-neighbor-based augmentation 并没有表现出稳定优势。</span></p><p data-tool="mdnice编辑器"><span leaf="">因此，如果只是做 scRNA-seq batch correction，可以优先把 </span><strong><span leaf="">scVI</span></strong><span leaf=""> 当作基线；如果目标是多模态整合或缺失模态预测，</span><strong><span leaf="">VICReg</span></strong><span leaf=""> 和 </span><strong><span leaf="">SimCLR</span></strong><span leaf=""> 反而是很好的起点。如果要使用 single-cell foundation model，最好不要只看 zero-shot，而是要认真评估 fine-tuning 后在具体任务上的表现。</span></p></section><section><span leaf=""><br></span></section><p><mp-style-type data-value="10000"></mp-style-type></p></div>
